python/ray/plasma/plasma_eventloop.py

```diff
@@ -0,0 +1,524 @@
+        results = []
+
+        for fut in self._children:
+            if fut.cancelled() and self.return_exceptions:
+                res = asyncio.futures.CancelledError()
+            elif fut._exception is not None and self.return_exceptions:
+                res = fut.exception()  # Mark exception retrieved.
+            else:
+                res = fut._result
+            results.append(res)
+
+        return results
+
+    def _done_callback(self, fut):
+        if self.done():
+            if not fut.cancelled():
+                # Mark exception retrieved.
+                fut.exception()
+            return
+
+        if not self.return_exceptions:
+            if fut.cancelled():
+                self.set_exception(asyncio.futures.CancelledError())
+                return
+            elif fut._exception is not None:
+                self.set_exception(fut.exception())
+                return
+
+        self.nfinished += 1
+
+        if self._halt_on():
+            self.set_result(self._collect_results())
+
+    def cancel(self):
+        if self.done():
+            return False
+        ret = False
+        for child in self._children:
+            if child.cancel():
+                ret = True
+        return ret
+
+    def flush_results(self):
+        done, pending = [], []
+        for f in self._children:
+            if f.done():
+                done.append(f)
+            else:
+                pending.append(f)
+            f.remove_done_callback(self._done_callback)
+        return done, pending
+
+    @asyncio.coroutine
+    def wait(self, timeout=None):
+        if not self._children:
+            return [], []
+
+        loop = self._loop
+
+        waiter = loop.create_future()
+        timeout_handle = None
+        if timeout is not None:
+            timeout_handle = loop.call_later(timeout, _release_waiter, waiter)
+
+        def _on_completion(_):
+            if timeout_handle is not None:
+                timeout_handle.cancel()
+            if not waiter.done():
+                waiter.set_result(None)
+
+        self.add_done_callback(_on_completion)
+
+        try:
+            yield from waiter
+        finally:
+            if timeout_handle is not None:
+                timeout_handle.cancel()
+
+        return self.flush_results()
+
+
+def gather(*coros_or_futures, loop=None, return_exceptions=False):
+    """This method resembles `asyncio.gather`.
+
+    Args:
+        *coros_or_futures: A list of coroutines or futures.
+        loop (PlasmaSelectorEventLoop): An eventloop.
+        return_exceptions (bool): If true, return exceptions as results
+            without raising them.
+
+    Returns:
+        PlasmaFutureGroup: A PlasmaFutureGroup object that
+        collects the results as a list.
+    """
+
+    fut = PlasmaFutureGroup(loop=loop, return_exceptions=return_exceptions)
+    for f in coros_or_futures:
+        fut.append(f)
+    return fut
+
+
+@asyncio.coroutine
```

Comment:
It is for compatibility. Python3.4 introduced `asyncio` module while Python3.5 legalized the `async` keyword. But you could be right because those who want to enjoy async features of python tends to use python3.5+.